client/src/context/reducers/chatReducer.js

```diff
@@ -40,7 +40,8 @@ export default function chatReducer(state, action) {
 
 		case 'CLOSE_CHAT': {
 			const { chatId } = action.payload;
-
+			localStorage.removeItem('cryptoKey' + chatId);
+			localStorage.removeItem('importPublicKey'+ chatId);
```

Comment:
why not use context rather than localstorage?

Reply:
I am using localStorage to store keys. Because they should remain same till the duration the chat exist. If i store it in context they would change if we reload or come back to web page again. If I dont use localStorage than i will not be able to decrypt older messages that exist in chat because they would have been encrypted with different key which is generated randomly for each chat. So i am clearing the localStorage when chat is removed or will not exist.

Reply:
good choice, thanks for explaining